### PR TITLE
Wri #916  external pubs

### PR DESCRIPTION
**Pull request by komejo**
## What issue(s) does this solve?
Authors (Person Content type) use variations of their names in print, so we are not showing all of their External Publications.

- [x] Issue Number: https://github.com/wri/wriflagship/issues/916

## What is the new behavior?
 - Adds two new fields to the Person content type: Alternative First Names, Alternative Last Names. 
 - Places those in a new tab: "Aliases"
 - Updates the external Pubs JSON to be a cached version ( https://files.wri.org/external-publications/externalpubs.json)
 - Modifies the code in `ExternalPubBlock.php` to look for all the available combination of names.
 - Also tweaks the layout of the External pubs so they correctly match the page width. 

## Profile requirements:

- Does deploying this change require a change to config at the site level (choose one)?
  - [ ] No config change is required.
  - [x] Yes, and I have written an update hook to apply these config changes.
  - [ ] Yes, and I've included updating instructions to be added to the release notes. The next release will need to be a major version increase. (Only do this in special cases.)

## Site-level pull requests for testing. Only merge when these PRs are approved:

Create or update any site-level pull requests following [the documentation](https://github.com/wri/WRIN/wiki/WRI-Dev-Workflow-(Thinkshout)#generating-a-multidev-for-wri_sites-work-review)

- [ ] Flagship PR: https://github.com/wri/wriflagship/pull/1214

## Checked on develop (TA to do)

- [ ] Brasil
- [ ] China
- [ ] Indonesia

### CODE REVIEW

File: modules/wri_person/config/install/core.entity_form_display.node.person.default.yml [hunk 1]
```diff
@@ -99,6 +106,40 @@ third_party_settings:
         formatter: open
         description: ''
         required_fields: true
+    group_publishing:
```
Comment (mariacha) [inline modules/wri_person/config/install/core.entity_form_display.node.person.default.yml:109]: This publishing group is WRI specific too, so might give you grief during site install. If not, you can leave it.

File: modules/wri_person/wri_person.install [hunk 2]
```diff
@@ -93,3 +93,29 @@ function wri_person_create_person_group() {
...
+function wri_person_update_10001() {
+  // Add the fields.
+  \Drupal::service('distro_helper.updates')->installConfig('field.storage.node.field_alt_first_names', 'wri_person', 'install', 'TRUE');
+  \Drupal::service('distro_helper.updates')->installConfig('field.field.node.person.field_alt_first_names', 'wri_person', 'install', 'TRUE');
+  \Drupal::service('distro_helper.updates')->installConfig('field.storage.node.field_alt_last_names', 'wri_person', 'install', 'TRUE');
+  \Drupal::service('distro_helper.updates')->installConfig('field.field.node.person.field_alt_last_names', 'wri_person', 'install', 'TRUE');
+
+  // Updating the Person form/view displays.
+  \Drupal::service('distro_helper.updates')->updateConfig('core.entity_form_display.node.person.default', [
+    'third_party_settings#field_group#group_more#group_aliases',
```
Comment (mariacha) [inline modules/wri_person/wri_person.install:109]: I think you don't need `group_more` in here? That should fix the missing Alias tab.

**Review by mariacha — CHANGES_REQUESTED** [review]
There's one PHPCS error, and I left a bunch of notes about things that will make the second test fail (installing from scratch). Those are all things you couldn't possibly know about, since they're Flagship vs. IO specific.

File: modules/wri_external_pub/src/Plugin/Block/ExternalPubBlock.php [hunk 3]
```diff
@@ -32,12 +32,14 @@ public function build() {
...
       && $node->bundle() === 'person') {
 
-      $pubListSrc = "https://s3.amazonaws.com/wriorg/external-publications/externalpubs.json";
+      $pubListSrc = "https://files.wri.org/external-publications/externalpubs.json";
       $pubList = file_get_contents($pubListSrc);
       $pubListArray = json_decode($pubList, TRUE);
 
       $authLast = $node->field_last_name->value;
       $authFirst = $node->field_first_name->value;
+      $authAltLast = $node->field_alt_last_names->value;
```
[the remark this answers is not in the document]
Comment (komejo) [reply modules/wri_external_pub/src/Plugin/Block/ExternalPubBlock.php:41]: Ahh, you're right - I got so excited it worked that I forgot to check multiple! I've updated it to loop through all of the potential combinations of First and Last names.

**Comment by komejo** [comment]
@mariacha W00t! 
Screenshot 2024-08-01 at 12 06 26 PM

**Review by mariacha — APPROVED** [review]

**Pull request merged**